### ISSUE TITLE
runtime: nodejs14.x support

### ISSUE THREAD

**Issue by kaihendry** [serverless/serverless#9980]
### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Use case description

manifest for lambci/lambda:nodejs14.x not found: manifest unknown: manifest unknown

when using starter `sls invoke local -f hello --docker`

### Proposed solution (optional)

Clearer documentation at the very least that 

```
$ grep runtime serverless.yml
  runtime: nodejs14.x
```

is unsupported. Thanks!

https://www.serverless.com/framework/docs/providers/aws/guide/services/

**Comment by medikoo** [comment]
@keihendry Support for `nodejs14.x` runtime was added a longer while ago (with https://github.com/serverless/serverless/pull/8894)

Is your report about outdated template? If so what exactly template? How did you install it?

**Comment by kaihendry** [comment]
The template I used was the first option from running `sls`. I updated the runtime the 14.x and the --docker option fails.

**Comment by medikoo** [comment]
@kaihendry thanks for the clarifications.

In terms of docker we depend on [docker-lambda](https://github.com/lambci/docker-lambda) project, and it appears that `nodejs14.x` runtime is not yet supported there: https://github.com/lambci/docker-lambda/issues/329

I'm still not sure why our template doesn't default to `nodejs14.x` but docker compatibility might be the reason. @mnapoli do you have any more information?

**Comment by mnapoli** [comment]
@medikoo no, but it seems like lambci's project is no longer active unfortunately. Since there are the SAM CLI docker images, it might make sense to migrate to those Docker images, but I have no idea how much work that represents.

**Comment by davidjb** [comment]
I did a write-up about the work required to resolve this: https://github.com/serverless/serverless/issues/10797#issuecomment-1544922990
